src/Uno.UI/UI/Xaml/Documents/TextFormatting/FontDetailsCache.skia.cs

```diff
@@ -1,59 +1,205 @@
+				{
+					if (_fontCache.TryGetValue(key, out var details))
+					{
+						if (typeface is null)
+						{
+							// font load failed.
+							details.LoadFailed();
+						}
+						else
+						{
+							details.Update(typeface);
+						}
+					}
+				}
+			}
+		}
+	}
+
+	private static string GetFamilyNameFromManifest(Stream jsonStream, FontWeight weight, FontStyle style, FontStretch stretch)
+	{
+		/*
+		 {
+			"fonts": [
+				{
+					"font_style": "normal",
+					"font_weight": 400,
+					"font_stretch": "normal",
+					"family_name": "ms-appx:///path/to/ExampleFontFamily-Regular.ttf"
+				},
+				{
+					"font_style": "italic",
+					"font_weight": "Normal",
+					"font_stretch": "normal",
+					"family_name": "ms-appx:///path/to/ExampleFontFamily-Italic.ttf"
+				},
+				{
+					"font_style": "normal",
+					"font_weight": 700,
+					"font_stretch": "normal",
+					"family_name": "ms-appx:///path/to/ExampleFontFamily-Bold.ttf"
+				}
+			]
+		} 
+		 */
+		var manifest = JsonSerializer.Deserialize<FontManifest>(jsonStream, _options);
+		if (manifest?.Fonts is null || manifest.Fonts.Length == 0)
+		{
+			throw new ArgumentException("Font manifest file is incorrect.");
+		}
+
+		var bestSoFar = manifest.Fonts[0];
+		for (int i = 1; i < manifest.Fonts.Length; i++)
+		{
+			var candidateMatch = manifest.Fonts[i];
+			if (candidateMatch.FontWeight != bestSoFar.FontWeight && Math.Abs(candidateMatch.FontWeight - weight.Weight) < Math.Abs(bestSoFar.FontWeight - weight.Weight))
+			{
+				// candidateMatch is a better match than bestSoFar. So it's now our new bestSoFar.
+				bestSoFar = candidateMatch;
+			}
+			else if (candidateMatch.FontStyle != bestSoFar.FontStyle && candidateMatch.FontStyle == style)
+			{
+				// The current bestSoFar
+				bestSoFar = candidateMatch;
+			}
+			else if (stretch != FontStretch.Undefined && candidateMatch.FontStretch != bestSoFar.FontStretch && Math.Abs(candidateMatch.FontStretch - stretch) < Math.Abs(bestSoFar.FontStretch - stretch))
+			{
+				// candidateMatch is a better match than bestSoFar. So it's now our new bestSoFar.
+				bestSoFar = candidateMatch;
+			}
+		}
+
+		return bestSoFar.FamilyName;
+	}
+
+	private static async Task<SKTypeface?> LoadTypefaceFromApplicationUriAsync(Uri uri, FontWeight weight, FontStyle style, FontStretch stretch)
+	{
+		try
+		{
+			// TODO (This comment should be resolved during code review):
+			// Should the user be responsible for adding ".manifest" to the ms-appx path himself?
+			// The benefit of the user doing so is that we will not have to first check if a manifest exists.
+			var manifestUri = new Uri(uri.OriginalString + ".manifest");
+			var manifestFile = await StorageFile.GetFileFromApplicationUriAsync(manifestUri);
+			var manifestStream = await manifestFile.OpenStreamForReadAsync();
+			uri = new Uri(GetFamilyNameFromManifest(manifestStream, weight, style, stretch));
+		}
+		catch
+		{
+			// manifest file is not found or cannot be read. Ignore it.
```

Comment:
We have an existing internal API to determine the presence of a file

Reply:
@jeromelaban What is it? I couldn't find it

Reply:
https://platform.uno/docs/articles/features/file-management.html#support-for-library-provided-assets: `StorageFileHelper.ExistsInPackage`

Reply:
@jeromelaban That's in Uno.UI.Toolkit 😕